utils/common.py

```
@@ -198,11 +203,74 @@ def get_hub_url():
+
+    auth_url = _hub_auth_url(hub_url)
+    basic = base64.b64encode(f"{username}:{password}".encode()).decode()
+    request = urllib.request.Request(
+        auth_url,
+        data=b"{}",
+        headers={
+            "Authorization": f"Basic {basic}",
+            "Content-Type": "application/json",
+        },
+        method="POST",
+    )
```

Ruff (0.15.18)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)